custom_components/moonraker/sensor.py

```diff
@@ -201,10 +201,50 @@ class MoonrakerSensorDescription(SensorEntityDescription):
 async def async_setup_entry(hass, entry, async_add_entities):
     """Setup sensor platform."""
     coordinator = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities([MoonrakerSensor(coordinator, entry, desc) for desc in SENSORS])
+
+    object_list = await coordinator.async_get_printer_object_list()
+    opt_sensors = await async_setup_optional_temp_sensors(object_list)
+
     async_add_entities(
-        [MoonrakerSensor(coordinator, entry, description) for description in SENSORS]
+        [MoonrakerSensor(coordinator, entry, desc) for desc in opt_sensors]
     )
 
+    coordinator.load_sensor_list(opt_sensors)
+
+
+async def async_setup_optional_temp_sensors(object_list):
+    """Setup optional sensor platform."""
+    sensor_list = []
+    for obj in object_list["objects"]:
+        split_obj = obj.split()
+        if split_obj[0] == "temperature_sensor":
+            sensor_list.append(
+                MoonrakerSensorDescription(
+                    key=split_obj[1],
+                    name=split_obj[1].replace("_", " ").title(),
+                    value_fn=MoonrakerOptionalTempSensor(obj),
+                    subscriptions=[(obj, "temperature")],
+                    icon="mdi:thermometer",
+                    unit=DEGREE,
+                )
+            )
+    return sensor_list
+
+
+class MoonrakerOptionalTempSensor:
```

Comment:
do we really need this class?
I think your proposed logic should be valuable for all sensors